BenadrylAutomation/src/main/java/com/jnj/pageobjects/ProductPage.java

```diff
@@ -0,0 +1,284 @@
+	}
+
+	/**
+	 * Verifies that the product overview section is visible on the product detail page.
+	 * @param expectedOverview The text expected to be found on the overview being verified.
+	 */
+	public void verifyProdOverview(String expectedOverview) {
+	    this.visitPDP();
+	    String actualOverview = prodOverviewTxt.getText();
+	    Action.printAndAssert(actualOverview, expectedOverview);
+	    extentInfoLog("Overview is displayed : ", Action.isDisplayed(driver, prodOverview));
+	}
+
+	/**
+	 * Verifies the presence and functionality of buttons on the product detail page, specifically 'Write a review' and 'Buy Now'.
+	 * @param expectedText The text expected to be found on the button being verified.
+	 * @param expectedModalText The text expected to be found on the heading of the modal.
+	 */
+	public void verifyButtons(String expectedText, String expectedModalText) {
+	    if (expectedText.contains("Write a review")) {
+	        this.verifyWriteAReviewBtn(expectedText);
+	    } else {
+	        this.verifyBuyNowBtn(expectedText, expectedModalText);
+	    }
+	}
+
+	/**
+	 * Verifies the 'Write a Review' button's presence, text, and click functionality on the product detail page.
+	 * @param expectedText The expected text to be displayed on the 'Write a Review' button.
+	 * @param expectedModalText The text expected to be found on the heading of the modal.
+	 */
+	public void verifyWriteAReviewBtn(String expectedText) {
+	    this.visitPDP();
+	    extentInfoLog("Button is displayed : ", Action.isDisplayed(driver, writeAReviewBtn));
+	    String actualText = writeAReviewBtn.getText();
+	    Action.printAndAssert(actualText, expectedText);
+	    Action.performActionwithExtentInfoLog(writeAReviewBtn, "click", "Clicking on : " + actualText);
+	    extentInfoLog("Write a Review modal is displayed : ", Action.isDisplayed(driver, writeAReviewModal));
+	}
+
+	/**
+	 * Verifies the 'Buy Now' button's presence, text, and click functionality on the product detail page.
+	 * @param expectedText The expected text to be displayed on the 'Buy Now' button.
+	 * @param expectedModalText The text expected to be found on the heading of the modal.
+	 */
+	public void verifyBuyNowBtn(String expectedBtnText, String expectedModalText) {
+	    this.visitPDP();
+	    extentInfoLog("Button is displayed : ", Action.isDisplayed(driver, buyNowBtn));
+	    String actualBtnText = buyNowBtn.getText();
+	    Action.printAndAssert(actualBtnText, expectedBtnText);
+	    Action.explicitWaitForElementTobeclickable(buyNowBtn, 30);
+	    Action.performActionwithExtentInfoLog(buyNowBtn, "click", "Clicking on : " + actualBtnText);
+	    Action.explicitWaitForElementTobeclickable(buyNowModalHeading, 30);
+	    String actualModalText = buyNowModalHeading.getText();
+	    Action.printAndAssert(actualModalText, expectedModalText);
+	    extentInfoLog("Buy Now modal is displayed : ", Action.isDisplayed(driver, buyNowModal));
```

Comment:
@zahinSJI no need of this. remove this line.

Reply:
Removed